Automattic/wp-e2e-tests
# Visdiffs 2.0: Integrating visdiffs with our e2e tests

This PR integrates our visdiff tests with our e2e tests so we can look for visual regressions in context on WordPress.com instead of the Calypso DevDocs. The visdiffs run and take screenshots from the following tests:

- e2e tests in the `specs` directory marked with the `@visdiff` suite tag
- a new spec in the `specs-visdiff` directory (where taking a screenshot in our e2e tests would lead to unreliable results)

These visdiffs currently work on OSX Chrome, Firefox, and Safari. They currently are not working in Win IE11 or Firefox mobile; due to time constraints we are going to launch the visdiffs as-is and work to enable them in additional browsers/screensizes in the future.

**Test with Local Screenshots:**

1. Set your local config, e.g. using a config file `local-visdiff.json` with `export NODE_ENV='visdiff'`
2. Run the visdiff specs in Sauce Labs: `./run.sh -v -l [browser] -s [screensize]`
3. View the resulting screenshots in the `/wp-e2e-tests/screenshots` directory.

**Test with Applitools Screenshot Diffs:**

1. Set your local config, e.g. using a config file `local-visdiff.json` with `export NODE_ENV='visdiff'`
2. Run the visdiff specs in Sauce Labs with Applitools (requires a CircleCI token): `sh scripts/trigger-circle-ci-specific.sh Automattic/wp-e2e-tests-visdiff master $CIRCLE_TOKEN '-v -l osx-chrome -s mobile,tablet,desktop' add/integrated-visdiffs`
3. Check Slack channel for notifications from Applitools diffs or visit Applitools dashboard directly for full results.
